pkg/controller/core/clusterqueue_controller.go

```diff
@@ -107,16 +110,17 @@ func (r *ClusterQueueReconciler) Reconcile(ctx context.Context, req ctrl.Request
 		}
 	}
 
-	status, err := r.Status(&cqObj)
-	if err != nil {
-		log.Error(err, "Failed getting status from cache")
-		return ctrl.Result{}, err
-	}
-
-	if !equality.Semantic.DeepEqual(status, cqObj.Status) {
-		cqObj.Status = status
-		err := r.client.Status().Update(ctx, &cqObj)
-		return ctrl.Result{}, client.IgnoreNotFound(err)
+	newCQObj := cqObj.DeepCopy()
+	if r.cache.ClusterQueueActive(newCQObj.Name) {
+		msg := fmt.Sprintf("ClusterQueue %q can admit new workloads", newCQObj.Name)
+		if err := r.UpdateCQStatusIfChanged(ctx, newCQObj, metav1.ConditionTrue, "Ready", msg); err != nil {
+			return ctrl.Result{}, client.IgnoreNotFound(err)
+		}
+	} else {
+		msg := fmt.Sprintf("ClusterQueue %q can't admit new workloads; The specified ResourceFlavors are not found", newCQObj.Name)
```

Comment:
```suggestion
		msg := "Can't admit new workloads; some flavors are not found"
```

But, IIRC, there are other possibilities, such as terminating.

Reply:
yes

Reply:
ok.
I will add the case for CQ is terminating.

Reply:
Done.
https://github.com/kubernetes-sigs/kueue/pull/415/commits/db3cb1315a731d8cc9974d09b92c2a41e68ea508